backend/src/main/java/heartbeat/service/report/GenerateReporterService.java

```diff
@@ -102,6 +130,70 @@ public static Map<String, String> getRepoMap(CodebaseSetting codebaseSetting) {
 		return repoMap;
 	}
 
+	private static List<BoardCSVConfig> getFixedBoardFields() {
+		List<BoardCSVConfig> fields = new ArrayList<>();
+		for (BoardCSVConfigEnum field : BoardCSVConfigEnum.values()) {
+			fields.add(BoardCSVConfig.builder().label(field.getLabel()).value(field.getValue()).build());
+		}
+		return fields;
+	}
+
+	private static String removeQuotes(String value) {
+		return value.replaceAll("\"", "");
+	}
+
+	public String getFieldDisplayValue(Object object) {
+		Gson gson = new Gson();
+		boolean isArray = false;
+		String result = "";
+		if (object instanceof JsonNull || object == null) {
+			return result;
+		}
+
+		if (object instanceof JsonPrimitive) {
+			object = removeQuotes(((JsonPrimitive) object).getAsString());
+		}
+
+		if (object instanceof Double || object instanceof String || object.toString().equals("{}")) {
+			return result;
+		}
+
+		if (object instanceof List && ((List<?>) object).isEmpty()) {
+			return "";
+		}
+
+		if (object instanceof List) {
+			isArray = true;
+			object = ((List<?>) object).get(0);
+		}
+
+		if (object != null) {
+			JsonObject jsonObject = gson.toJsonTree(object).getAsJsonObject();
+			JsonElement name = jsonObject.get("name");
+			if (name != null) {
+				result = ".name";
+			}
+			JsonElement displayName = jsonObject.get("displayName");
+			if (displayName != null) {
+				result = ".displayName";
+			}
+			JsonElement value = jsonObject.get("value");
+			if (value != null) {
+				result = ".value";
+			}
+			JsonElement key = jsonObject.get("key");
+			if (key != null) {
+				result = ".key";
+			}
+		}
+
+		if (isArray) {
+			result = "[0]" + result;
+		}
+
+		return result;
+	}
+
 	public synchronized ReportResponse generateReporter(GenerateReportRequest request) {
```

Comment:
why add `synchronized` for this fuction?

Reply:
This `synchronized` was added before, not in this PR. 
The reason is front end will send two report requests at the same time.